doc/sphinx/source/immovable_program.rst

```diff
@@ -0,0 +1,150 @@
+.. _immovable_programs:
+
+Immovable Program Reference
+===========================
+Immovables can have programs that will be executed by the game engine. It is not mandatory for immovables to define programs. Immovables without a program will simply display their main animation indefinitely. Programs are required to allow workers to interact with an immovable (e.g. a tree will need a "fall" program to allow woodcutters to remove the tree).
+
+Programs are defined as Lua tables. Each program must be declared as a subtable in the immovable's Lua table called ``programs`` and have a unique table key. The entries in a program's subtable are the ``actions`` to execute, like this::
+
+   programs = {
+      program = {
+         "animate=idle 1550000",
+         "transform=deadtree4 5",
+         "seed=alder_summer_sapling 180",
+      },
+      fall = {
+         "remove=",
+      },
+   },
+
+The different command types and the parameters that they take are explained below.
+
+.. highlight:: default
+
+Command Types
+^^^^^^^^^^^^^
+- `animate`_
+- `transform`_
+- `grow`_
+- `remove`_
+- `seed`_
+- `construct`_
+- `playsound`_
+
+
+animate
+-------
+Runs an animation.
+
+Parameter syntax::
+
+  parameters ::= animation duration
+
+Parameter semantics:
+
+``animation``
+    The name of an animation (defined in the productionsite).
+``duration``
+    A natural integer specifying the duration in milliseconds.
+
+Starts the specified animation for the productionsite. Blocks the execution of the program for the specified duration. (The duration does not have to equal the length of the animation. It will loop around. The animation will not be stopped by this command. It will run until another animation is started.)
+
+transform
+---------
+Delete this immovable and instantly replace it with a different immovable.
+
+Parameter syntax::
+
+  parameters ::= [bob] name [chance]
+
+Parameter semantics:
+
+``name``
+    The name of the immovable to turn into. If the ``bob`` flag is given, this refers to any kind of bob; otherwise to an immovable.
+``chance``
+    A natural integer in [1,254] defining the chance that the transformation succeeds. If it doesn't succeed, the next program step is triggered. If ``chance`` is omitted, the transformation will always succeed.
```

Comment:
We should translate the integer to % here

Reply:
I don't understand, the engine uses values from 0 to 255 so there is no percentage translation possible?

Reply:
first we should update the text then that it reflects 0, 255.
second I thought about having a second bracket stating [0%, 99,5%] is equivalent to [0, 255]. or a statement that each integerinkrement represents roughly 0,4%

Reply:
The corner values 0 and 255 are forbidden because 0 would mean that it never succeeds (in which case the statement is superfluous), and 255 would mean certain success (in which case `chance` should be omitted instead).

So the text should be rephrased somewhat like this:
`
A natural integer in [1,254] defining the chance that the transformation succeeds. This equals a range from [0%,100%] with a difference of 1 unit equalling (100/255)% (approximately 0.39%). If the step doesn't succeed, the next program step is triggered. If »chance« is omitted, the transformation will always succeed.
`
In my opinion this would leave it unclear to the reader whether we are using % or 255th's, and cause confusion whether % are also allowed or why such conversions are necessary at all…

Reply:
Well, as this is directed at personnel which might want to implement some immovable what about giving the formula used by the program to make it clear what impact a value set here has. 
This was my only concern: giving the reader a chance to choose the right value for its intention.

Reply:
Okay, like this?